client/signup/config/steps-pure.js

```diff
@@ -101,7 +101,7 @@ export function generateSteps( {
 			stepName: 'domains-launch',
 			apiRequestFunction: addDomainToCart,
 			fulfilledStepCallback: isDomainFulfilled,
-			providesDependencies: [ 'domainItem' ],
+			providesDependencies: [ 'domainItem', 'shouldHideFreePlan' ],
```

Comment:
Hi @jdc91 :wave: This change is different from what was in #44961, is that right?

In #44961, the `shouldHideFreePlan` was added to `dependencies`, not to `providesDependencies`. But that was making the flow dependencies invalid and the [_assertFlowHasValidDependencies](https://github.com/Automattic/wp-calypso/blob/2f156ae44e2c4d6151699e8e253e5d93d0657cc6/client/lib/signup/flow-controller.ts#L165) check would fail.

That's because the `domains-launch` step depends on the `shouldHideFreePlan` value, but it's not provided by anyone -- not in the query string (like `siteSlug`), nor by any other step in the flow. The `domains-launch` step actually _provides_ the value for other steps to consume.

The result of the failed validation is that the `launch-site` flow is redirected to the default flow (`/start`), breaking completely away from the expected UI.

The visible result is that instead of seeing `domains-launch` UI with a "Skip purchase" option, as the e2e test expects:

<img width="999" alt="Screenshot 2020-09-15 at 10 30 37" src="https://user-images.githubusercontent.com/664258/93194618-948a0280-f748-11ea-851f-21a1f9dcbaa3.png">

I am redirected to `/start` and then to `/start/domains` (the first step of the default flow), seeing this:

<img width="778" alt="Screenshot 2020-09-15 at 11 44 16" src="https://user-images.githubusercontent.com/664258/93194852-d5821700-f748-11ea-9716-14e8c962f0b1.png">

That's exactly what we've seen in the failed e2e tests.

The `_assertFlowHasValidDependencies` check verifies if the dependency is present in the dependency store, which can be loaded from local storage on boot. That explains the intermittency of the failures that @scinos observed: sometimes `shouldHideFreePlan` was already in local storage from previous run of the `domains-launch` step and the validation succeeded.

Reply:
Yes @jsnajdr this was the breaking change as mentioned above by @niranjan-uma-shankar. Thanks for the lengthy explanation why it was intermittent. 👍

Reply:
I still don't fully understand why the 'launch site' e2e test failed even after #44961 was reverted, but I suspect it has something to do with the conditional submission of the `{ shouldHideFreePlan }` dependency. That seems wrong and should be fixed before merging.

I looked at the e2e failures in this PR and they seem completely unrelated. Related to Media and Gutenberg. No problem merging the PR despite them.